pytest_mh/utils/tc.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from .. import MultihostHost, MultihostUtility
+from ..ssh import SSHLog, SSHProcessError
+
+__all__ = ["LinuxTrafficControl"]
+
+
+class LinuxTrafficControl(MultihostUtility):
+    """
+    Perform traffic control operations on remote host.
+
+    All changes are automatically reverted when a test is finished.
+    """
+
+    def __init__(self, host: MultihostHost) -> None:
+        """
+        :param host: Remote host instance.
+        :type host: MultihostHost
+        """
+        super().__init__(host)
+        self.__restore = set()  # type: set[str]
+
+        # Let's find out the interface that connects the systems
+        self.__interface: str = ""
+        result = self.host.ssh.run(f"dig +short {host.hostname}", log_level=SSHLog.Error)
+        try:
+            result = self.host.ssh.run(f"ip -o address | grep {result.stdout}", log_level=SSHLog.Error)
+            self.__interface = result.stdout.split(" ")[1]
+        except SSHProcessError:
+            print(f"{host.hostname} doesn't provide 'ip' command")
+
+    def teardown_when_used(self):
+        """
+        Revert all firewall changes.
+
+        :meta private:
+        """
+        for iter in self.__restore:
+            self.host.ssh.run(iter, log_level=SSHLog.Error)
```

Comment:
It would be better to run all commands at once to avoid running multiple ssh channels.

Reply:
Fixed